### PR TITLE
cpu: samd21: misc compile fixes

### PR DESCRIPTION
**Pull request by kaspar030**
This PR makes SAM R21 unittests compile with -pedantic.

### CODE REVIEW

**Comment by kaspar030** [comment]
@OlegHahm I re-introduced the fixed-size array you dropped in #4358. IMHO -pedantic compiling is more important than automatic array size counting for a file we almost never change, don't you think?

**Comment by miri64** [comment]
Maybe we can merge this for the release?

**Comment by kaspar030** [comment]
@OlegHahm I think the discussion about the size literal for the vectors array was sth that we didn't like?

File: cpu/samd21/vectors.c [hunk 1]
```diff
@@ -66,54 +66,59 @@ WEAK_DEFAULT void isr_ptc(void);
 WEAK_DEFAULT void isr_i2c(void);
 
 /* interrupt vector table */
-ISR_VECTORS const void *interrupt_vector[] = {
+ISR_VECTORS struct {
+    void* _estack;
+    void(*vectors[43])(void);
```
Comment (OlegHahm) [inline cpu/samd21/vectors.c]: Can you replace this magic number with a macro?

**Comment by OlegHahm** [comment]
I found it a somewhat ugly solution, yes. But since the number of interrupts for a certain MCU is not gonna chance, it's certainly a viable solution.

**Comment by kaspar030** [comment]
- rebased
- use macro instead of literal

**Comment by PeterKietzmann** [comment]
ACK but unmergable

**Comment by kYc0o** [comment]
Please rebase. Why unmergeable?

**Comment by PeterKietzmann** [comment]
This branch has conflicts that must be resolved

Conflicting files
cpu/cortexm_common/cortexm_init.c

**Pull request merged**